plugins/catalog-backend/src/ingestion/HigherOrderOperations.ts

```diff
@@ -91,19 +88,34 @@ export class HigherOrderOperations implements HigherOrderOperation {
     // in the entities list. But we aren't sure what to do about those yet.
 
     // Write
-    if (!previousLocation) {
-      await this.locationsCatalog.addLocation(location);
-    }
-    const outputEntities: Entity[] = [];
-    for (const entity of readerOutput.entities) {
-      const out = await this.entitiesCatalog.addOrUpdateEntity(
-        entity.entity,
-        location.id,
-      );
-      outputEntities.push(out);
-    }
+    const entities = await this.database.transaction(async tx => {
```

Comment:
The higher order operation uses multiple database calls (matches it's name? 😆 ). That means we have to move the transaction up to here.

Reply:
Myeah ... it's a bit unfortunate but I don't actually have a better solution to offer right now. May want to refactor away the entire HigherOrderOperations class at some point but not right now.